Traefik: Allow configuration of source IP ranges in load balancer.

octete opened the pull request:
This PR will allow to configure traefik's service to restrict its access to a list of source IP ranges, if it uses the service type `LoadBalancer` and if it is supported.

Please, let me know if this satisfies all requirements and style. If not, I'll change the PR.

Thanks.

octete committed:
Allow loadBalancerSourceRanges

This change allows us to specify loadBalancerSourceRanges for when we
are using loadBalancerIP in the service definition.

octete committed:
Add documentation for loadBalancerSourceRanges

dhilipkumars commented:
/ok-to-test

dhilipkumars commented on stable/traefik/values.yaml:
```diff
@@ -3,6 +3,7 @@ image: traefik
 imageTag: 1.3.4
 serviceType: LoadBalancer
 loadBalancerIP:
+#loadBalancerSourceRanges: []
```
Now that we have more than one `value` that starts with loadBalancer, could we
```
loadBalancer:
   IP:
   sourceRanges: []
```

octete replied:
Of course. I can rewrite the PR if you wish. I was looking at how other charts were doing this same thing (i.e. prometheus) and they seem to have a flat structure. 

I'll commit the changes.

prydonius replied:
I think this is okay for now given the flat structure of `image` `imageTag`

octete committed:
Change to a nested structure for load balancer

Now that we can specify both the IP and an optional list of source IPs
for the load balancer setting, make it nested.

dhilipkumars commented:
/assign
LGTM from my side
cc:@prydonius could you PTAL and approve its if its all fine.

krancour commented on stable/traefik/README.md:
```diff
@@ -88,7 +88,8 @@ The following tables lists the configurable parameters of the Traefik chart and
 | `imageTag`                      | The version of the official Traefik image to use                     | `1.3.1`                                  |
 | `serviceType`                   | A valid Kubernetes service type                                      | `LoadBalancer`                            |
-| `loadBalancerIP`                | An available static IP you have reserved on your cloud platform      | None                                      |
+| `loadBalancer.IP`               | An available static IP you have reserved on your cloud platform      | None                                      |
```
This is a breaking change.

krancour replied:
So, although this is cleaner, I wouldn't suggest this.

unguiculus replied:
This is an unnecessary breaking change. It should be `loadBalancerIP` and `loadBalancerSourceRanges`. IMHO, this additional level of nesting is just too much. Also, values should start with a lower case letter.

Please see chart examples for reference:
https://github.com/kubernetes/helm/blob/master/docs/examples/nginx/templates/service.yaml#L22-L28
https://github.com/kubernetes/helm/blob/master/docs/examples/nginx/values.yaml#L24-L25

octete replied:
@unguiculus That was the original PR. @dhilipkumars asked me to change it to add nesting, so that went in 68324a7f835e1b0beec1dbe4e5bb5a2ca98c595d. 
I can revert that commit and leave it as it was originally intended.

Would that be OK with everyone?

Thank you.

dhilipkumars replied:
It is not @octete fault, it was me who requested this.  Clearly, I did not think through the breaking change part.  

Apologies to @octete for the confusion caused `(especially on your first PR to this repo)`, please revert it to flat structure.

octete replied:
No problem at all. Thanks @dhilipkumars, I will revert and commit the change straight away.

Thanks all.

octete committed:
Revert "Change to a nested structure for load balancer"

This reverts commit 68324a7f835e1b0beec1dbe4e5bb5a2ca98c595d.
This was a breaking change, so we are leaving as a flat structure as the
original PR.

krancour commented:
LGTM
